addons/binding/org.openhab.binding.te923/src/main/java/org/openhab/binding/te923/handler/Te923Handler.java

```diff
@@ -0,0 +1,217 @@
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.te923.Te923BindingConstants;
+import org.openhab.binding.te923.Te923Channel;
+import org.openhab.binding.te923.internal.ExecUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Te923Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Gabriel Klein - Initial contribution
+ */
+@NonNullByDefault
+public class Te923Handler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(Te923Handler.class);
+
+    // don't "spam" the Te923 command. Max one call per 3 secs
+    private static final long MIN_UPDATE_TIME = 3000;
+
+    // refresh frequency
+    private long refresh = 60;
+
+    private String cmdte923con = Te923BindingConstants.DEFAULT_CMD;
+
+    @Nullable
+    private ScheduledFuture<?> refreshJob;
+
+    public Te923Handler(Thing thing) {
+        super(thing);
+    }
+
+    private double[] te923Data = new double[22];
+    private long lastTE923Update = 0;
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        logger.debug("Received channel: {}, command: {}", channelUID, command);
+
+        // No real commands
+
+    }
+
+    @Override
+    public void initialize() {
+
+        // updateStatus(ThingStatus.INITIALIZING);
+
+        Configuration config = getThing().getConfiguration();
+
+        try {
+            refresh = Long.parseLong("" + config.get("refresh"));
+        } catch (Exception e) {
+            logger.debug("Cannot set refresh parameter.", e);
+        }
+
+        if (refresh < 5) {
+            logger.debug("Cannot set refresh parameter to a value lower than 5");
+            refresh = 5;
+        }
+
+        try {
+            Object o = config.get("cmdte923con");
+            if (o == null) {
+                o = "";
+            }
+            cmdte923con = "" + o;
+        } catch (Exception e) {
+            logger.debug("Cannot set cmdte923con parameter.", e);
+        }
+
+        // Check that cmdte923 exists
+        {
+            if (cmdte923con == null || cmdte923con.length() == 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Location of te923con is not configured. More information on http://te923.fukz.org/");
+                return;
+            }
+
+            double d[] = getTe923Data();
+            if (d == null) {
+                return;
+            }
+
+        }
+
+        startAutomaticRefresh();
+
+    }
+
+    /**
+     * Get Te923 Data. Null if no data.
+     * This command update the status of this component.
+     *
+     * @return
+     */
+    private double @Nullable [] getTe923Data() {
+
+        // avoid spamming the system.
+        if (System.currentTimeMillis() - lastTE923Update < MIN_UPDATE_TIME) {
```

Comment:
Why keep track of the time? If you only call the method from the scheduler, the scheduler will guaranty it will only be called at the scheduled time.